docs/apache-airflow/core-concepts/params.rst

```diff
@@ -47,7 +47,16 @@ Use a dictionary that maps Param names to either a :class:`~airflow.models.param
             "x": Param(5, type="integer", minimum=3),
             "my_int_param": 6
         },
-    ):
+    ) as dag:
+
+.. note::
+
+   DAG-level parameters are the default values passed on to tasks. These should not be confused with values manually
+   provided through the UI form or CLI, which exist solely within the context of a :class:`~airflow.models.dagrun.DagRun`
+   and a :class:`~airflow.models.taskinstance.TaskInstance`. This distinction is crucial for TaskFlow DAGs, which may
+   include logic within the ``with DAG(...) as dag:`` block. In such cases, users might try to access the manually-provided
+   parameter values using the ``dag`` object, but this will only ever contain the default values. To ensure that the
+   manually-provided values are accessed, use a template variable such as ``params`` or ``ti`` within your task.
```

Comment:
NIT. Maybe add an example ?

Reply:
I do have a suitable example, just wondering if I should modify the snippet directly above this comment, or add a new one.

```python
from airflow import DAG
from airflow.decorators import task
from airflow.models.param import Param

with DAG(
    "the_dag",
    params={
        "x": Param(5, type="integer", minimum=3),
        "my_int_param": 6
    },
) as dag:
    
    @task.python
    def example_task(params: dict):
        # This will print the default value, 6:
        dag.log.info(dag.params['my_int_param'])
        # This will print the manually-provided value, 4:
        dag.log.info(params['my_int_param'])
        
    example_task()
        
if __name__ == "__main__":
    dag.test(
        run_conf={"my_int_param": 4}
    )

```

Reply:
Yeah. Adding it to the snippet above is a good idea :). Soimetimes good example is worth >>> description.

Reply:
Done